### PR TITLE
feat(autocomplete): add Autocomplete v1 template

### PR DESCRIPTION
**Pull request by francoischalifour**
This adds a template for Autocomplete v1.

## Changes

- The Autocomplete.js template was renamed "Autocomplete.js 0" to avoid confusion (and it still doesn't appear in the CLI)
- This new template is named "Autocomplete"

## Preview

Sandbox preview

You can see a preview of what CISA generates in this [**sandbox →**](https://codesandbox.io/s/8bmy6).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -127,8 +127,11 @@ jobs:
           template: 'Angular InstantSearch'
           app: 'angular-instantsearch'
       - build_app:
-          template: 'Autocomplete.js'
+          template: 'Autocomplete.js 0'
           app: 'autocomplete.js'
+      - build_app:
+          template: 'Autocomplete'
+          app: 'autocomplete'
```
Comment (francoischalifour) [inline .circleci/config.yml:134]: The ESLint version used in the template is not compatible with Node 10 so I only added the CI tests to Node 12.

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-12.14.1
+14.15.0
```
Comment (francoischalifour) [inline .nvmrc:1]: A Node bump is required for Parcel 2.

File: .circleci/config.yml [hunk 3]
```diff
@@ -97,7 +97,7 @@ jobs:
       - run: *run_yarn_install
       - save_cache: *save_yarn_cache
       - build_app:
-          template: 'Autocomplete.js'
+          template: 'Autocomplete.js 0'
```
Comment (sarahdayan) [inline .circleci/config.yml:100]: Maybe "v0" or "v0.x" would be more explicit.

File: src/templates/Autocomplete/.template.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const install = require('../../tasks/node/install');
+const teardown = require('../../tasks/node/teardown');
+
+module.exports = {
+  category: 'Web',
+  libraryName: '@algolia/autocomplete-js',
+  templateName: 'autocomplete',
+  appName: 'autocomplete-app',
+  keywords: ['algolia', 'Autocomplete', 'autocomplete'],
```
Comment (sarahdayan) [inline src/templates/Autocomplete/.template.js:9]: Is it necessary to have two casing variations?

File: src/templates/Autocomplete/app.js.hbs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const searchClient = algoliasearch('{{appId}}', '{{apiKey}}');
+
+autocomplete({
+  container: '#autocomplete',
+  placeholder: '{{placeholder}}',
+  debug: true,
+  getSources({ query }) {
+    return [
+      {
+        sourceId: 'products',
```
Comment (sarahdayan) [inline src/templates/Autocomplete/app.js.hbs:17]:
```suggestion
        sourceId: 'items',
```

File: src/templates/Autocomplete/app.js.hbs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                        attribute="{{attributesToDisplay.[0]}}"
+                      />
+                    </div>
+                  </div>
+                </div>
+              </div>
+            );
+          },
+          noResults() {
+            return 'No products matching.';
```
Comment (sarahdayan) [inline src/templates/Autocomplete/app.js.hbs:47]:
```suggestion
            return 'No matching items.';
```

File: src/templates/Autocomplete/index.html.hbs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <link rel="shortcut icon" href="favicon.png" type="image/x-icon" />
+    <link rel="stylesheet" href="style.css" />
+
+    <title>Algolia Starter | Autocomplete</title>
```
Comment (sarahdayan) [inline src/templates/Autocomplete/index.html.hbs:9]:
```suggestion
    <title>{{name}}</title>
```

File: src/templates/Autocomplete/index.html.hbs [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
```
Comment (sarahdayan) [inline src/templates/Autocomplete/index.html.hbs:5]: We should have their app ID at this stage, right? If yes, we could add a `<link rel="preconnect">` element to further improve the bootstrapped app.

File: .circleci/config.yml (hunk 3 above)
Comment (francoischalifour) [reply .circleci/config.yml:100]: We use this pattern for other templates too. Let's rename all of them in a major?

File: src/templates/Autocomplete/.template.js (hunk 4 above)
Comment (francoischalifour) [reply src/templates/Autocomplete/.template.js]: Probably not—removed in c65154c99b967dcc44907b4e826c4b0c0c2aa8fd.

File: src/templates/Autocomplete/app.js.hbs (hunk 5 above)
Comment (francoischalifour) [reply src/templates/Autocomplete/app.js.hbs]: See aa2a66f819db94207050bafe8a066330967af815.

File: src/templates/Autocomplete/app.js.hbs (hunk 6 above)
Comment (francoischalifour) [reply src/templates/Autocomplete/app.js.hbs]: See aa2a66f819db94207050bafe8a066330967af815.

File: src/templates/Autocomplete/index.html.hbs (hunk 7 above)
Comment (francoischalifour) [reply src/templates/Autocomplete/index.html.hbs]: See 39ec30ca6914975a146bfa371bf5c54b7d044aa5.

File: src/templates/Autocomplete/index.html.hbs (hunk 8 above)
Comment (francoischalifour) [reply src/templates/Autocomplete/index.html.hbs:5]: We mostly use templates on CodeSandbox, where we modify credentials after templates have been built for bug reproduction. So most of the time, the `<link rel="preconnect">` will not match the actual creds used.

File: scripts/release-templates.js [hunk 9]
```diff
@@ -71,13 +71,11 @@ async function build() {
         appId: 'latency',
         apiKey: '6be0576ff61c053d5f9a3225e2a90f76',
         indexName:
-          libraryName === 'autocomplete.js'
+          libraryName === 'autocomplete'
```
Comment (Haroenv) [inline scripts/release-templates.js:74]: Technically a user could manually have written a template, making this a breaking change, but I don't think it's a worry

File: src/templates/Autocomplete/index.html.hbs (hunk 8 above)
Comment (Haroenv) [reply src/templates/Autocomplete/index.html.hbs:5]: if we add this, it should be done for all templates, and maybe even programmatically in the js client

**Review by Haroenv — APPROVED** [review]
works for me!

File: .circleci/config.yml (hunk 3 above)
Comment (sarahdayan) [reply .circleci/config.yml:100]: Sure!

**Review by sarahdayan — APPROVED** [review]

File: scripts/release-templates.js (hunk 9 above)
Comment (francoischalifour) [reply scripts/release-templates.js]: You're right, we don't need this change anyway.

**Pull request merged**